### ISSUE TITLE
[Bug]: plotting error associated with Dirichlet-multinomial likelihood applied to generalized size frequency data

### ISSUE THREAD

**Issue by iantaylor-NOAA** [r4ss/r4ss#898]
### Describe the bug

As reported by @e-perl-NOAA in https://github.com/nmfs-ost/ss3-source-code/pull/545#issuecomment-1877234262, `SS_plots()` exits with the error:
```
Starting age comp data plots (group 14)
Error in if (CompError == 1) { : missing value where TRUE/FALSE needed
```

### To Reproduce

Run this test model with SS3 version 3.30.22, try `SS_plots()`.

### Expected behavior

Success plotting. 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/r4ss/r4ss#code-of-conduct)

**Comment by e-perl-NOAA** [comment]
@iantaylor-NOAA this is specific to the new test model that we added - Simple_with_DM_sizefreq. Probably something with the length_data_info implementing Dirichlet-Multinomial

**Comment by iantaylor-NOAA** [comment]
Refactoring of the r4ss code associated with the partition-specific data-weighting in early 2023 via https://github.com/r4ss/r4ss/pull/766 failed to read all the needed output for D-M likelihoods applied to size comp data. This should now be fixed via 5dfab83. I will do a little more testing tomorrow and see how the github actions turn out.
